### PR TITLE
Webpack config

### PR DESCRIPTION
**Pull request by Rostomjan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "build": "ng build && node server.js",
+    "build": "ng build && npm run webpack && node dist/server.js",
+    "webpack": "webpack --config webpack.config.js",
+    "webpack-w": "webpack --config webpack.config.js --hot --watch",
```
Comment (mykytamorachov) [inline package.json]: Please name task correctly this name means nothing

File: sh.exe.stackdump [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Exception: STATUS_STACK_OVERFLOW at rip=000772FD527
```
Comment (mykytamorachov) [inline sh.exe.stackdump]: Wtf is this?)

File: variables.env [hunk 3]
```diff
@@ -0,0 +1,10 @@
+NODE_ENV=development
```
Comment (mykytamorachov) [inline variables.env]: Do not please store such data in environment variables, it's not secure. Just create file with constants

**Review by mykytamorachov — CHANGES_REQUESTED** [review]

File: sh.exe.stackdump (hunk 2 above)
Comment (Rostomjan) [reply sh.exe.stackdump]: Omg It was a damp file(

File: server/constants/consts.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const config = {
```
Comment (mykytamorachov) [inline server/constants/consts.ts]: First of all, it's database config, so could you please name both file and constant in the way we can understand this is database config

File: package.json [hunk 5]
```diff
@@ -5,7 +5,10 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "build": "ng build && node server.js",
+    "build": "ng build && npm run webpack",
+    "server": "node dist/server.js",
+    "webpack": "webpack --config webpack.config.js",
```
Comment (mykytamorachov) [inline package.json]: What does this name mean?

try something like build-server and build-server-watch for these commands to make some actual sense

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+const path = require('path');
```
Comment (mykytamorachov) [inline webpack.config.js]: this filename should not be webpack config, cause when you call just webpack it looks for default config with exactly this name. Please change it to webpack.server.config.js

File: package.json (hunk 5 above)
Comment (mykytamorachov) [reply package.json]: And please do not forget to change the name of webpack file as it will be changed in comment below

File: package.json [hunk 7]
```diff
@@ -23,7 +26,7 @@
     "@angular/platform-browser-dynamic": "^4.2.4",
     "@angular/router": "^4.2.4",
     "body-parser": "^1.17.2",
-    "core-js": "^2.4.1",
+    "core-js": "^2.4.1",
```
Comment (mykytamorachov) [inline package.json]: Please remove empty whitespace.

File: server/constants/consts.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+    DATABASE_LOCAL: 'mongodb://127.0.0.1:27017/scafeinder',
+    DATABASE_CLOUD: '',
+    PORT: ''
+};
```
Comment (mykytamorachov) [inline server/constants/consts.ts]: Please add empty line in the end of file.

File: src/tsconfig.server.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    "target": "es5",
+    "baseUrl": "",
+    "types": [
+      "node"
+    ]
+  },
+  "include": [
+    "src/server/**/*.ts"
+  ]
+}
```
Comment (mykytamorachov) [inline src/tsconfig.server.json]: Same here

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        process: true,
+        Buffer: true
+    },
+
+    output: {
+        path: path.join(process.cwd(), 'dist'),
+        filename: '[name].js',
+        chunkFilename: '[id].chunk.js'
+    },
+};
```
Comment (mykytamorachov) [inline webpack.config.js]: and here

File: server/server.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+app.set("port", CONSTS.config.PORT || 3000);
+
+
+/**
+ * Listen on provided port, on all network interfaces.
+ */
+app.listen(app.get("port"), () => {
+  console.log(("  App is running at http://localhost:%d in %s mode"), app.get("port"), app.get("env"));
+  console.log("  Press CTRL-C to stop\n");
+});
```
Comment (mykytamorachov) [inline server/server.ts]: and here

File: server/routes/api.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import {Response, Request} from "express";
+
+/* GET api listing. */
+export let getApi = (req: Request, res: Response) => {
+  res.send('api works');
+  };
```
Comment (mykytamorachov) [inline server/routes/api.ts]: Add empty line in the end of file please

**Review by mykytamorachov — CHANGES_REQUESTED** [review]

File: server/constants/consts.ts (hunk 8 above)
Comment (MaksymI) [reply server/constants/consts.ts]: done

**Review by mykytamorachov — APPROVED** [review]

**Pull request merged**